Source/astcenccli_toplevel.cpp

```diff
@@ -1163,43 +1134,53 @@ int main(
 			break;
 	}
 
-	int array_size = 1;
-	int silentmode = 0;
-	int y_flip = 0;
-	int linearize_srgb = 0;
-	int thread_count = 0;		// default value
-
-	int low_fstop = -10;
-	int high_fstop = 10;
-
 	// TODO: Handle RAII resources so they get freed when out of scope
 	// Load the compressed input file if needed
 
 	// This has to come first, as the block size is in the file header
 	astc_compressed_image image_comp;
 	if (op_mode == ASTC_DECODE)
 	{
-		int error = load_astc_file(input_filename.c_str(), image_comp);
+		error = load_astc_file(input_filename.c_str(), image_comp);
 		if (error)
 		{
 			return 1;
 		}
 	}
 
-	astcenc_config codec_configuration;
-	if(init_astcenc_config(argc, argv, profile, op_mode, image_comp, codec_configuration))
+	astcenc_config config;
+
+    error = init_astcenc_config(argc, argv, profile, op_mode, image_comp, config);
+	if(error)
 	{
 		printf("ERROR: Astcenc configuration intialization failed.\n");
 		return 1;
 	}
 
-	if(edit_astcenc_config(argc, argv, op_mode,	array_size, silentmode, y_flip, linearize_srgb, thread_count, low_fstop, high_fstop, codec_configuration))
+	astcenc_swizzle swz_encode {
+		ASTCENC_SWZ_R,
+		ASTCENC_SWZ_G,
+		ASTCENC_SWZ_B,
+		ASTCENC_SWZ_A
+	};
+
+	astcenc_swizzle swz_decode {
+		ASTCENC_SWZ_R,
+		ASTCENC_SWZ_G,
+		ASTCENC_SWZ_B,
+		ASTCENC_SWZ_A
+	};
+
+	cli_config cli_config_options { 1, 0, 0, 0, 0, -10, 10 }; // Set default values
```

Comment:
Nit - I'd swap these two - call the type "cli_config_options", and call the variable "cli_config" (less typing on use)

Reply:
done